src/platforms/common/blackpill-f4/blackpill-f4.c

```diff
@@ -291,6 +291,48 @@ uint8_t platform_spi_xfer(const spi_bus_e bus, const uint8_t value)
 	}
 }
 
+void platform_spi_xfer_block(const spi_bus_e bus, uint8_t *const data, const size_t count)
+{
+	uint32_t spi_base = OB_SPI;
+	switch (bus) {
+	case SPI_BUS_INTERNAL:
+		spi_base = OB_SPI;
+		break;
+	case SPI_BUS_EXTERNAL:
+		spi_base = EXT_SPI;
+		break;
+	default:
+		return;
+	}
+
+#if 0
```

Comment:
No, it can't, because it blocking-waits for the entire duration of 8/16-bit SPI word in https://github.com/libopencm3/libopencm3/blob/201f5bcfb3fa70ee34818152463e7139f24db377/lib/stm32/common/spi_common_all.c#L189-L190
But thanks to that it does not submit an extra word in flight to keep data pumping, and hence cannot miss an Rx byte.

Reply:
ok, fair enough - then drop the simpler loop please as there's no point keeping it in this new code.